service/lib/dinstaller/dbus/question.rb

```diff
@@ -0,0 +1,172 @@
+          #
+          # @note Clients are responsible of generating the proper label for each option.
+          #
+          # @return [Array<String>]
+          def options
+            backend.options.map(&:to_s)
+          end
+
+          # Default option a client should offer as answer
+          #
+          # @return [String]
+          def default_option
+            backend.default_option.to_s
+          end
+
+          # Answer selected for a client
+          #
+          # @return [String]
+          def answer
+            backend.answer.to_s
+          end
+
+          # Selects an option as answer
+          #
+          # @param option [String]
+          def answer=(option)
+            backend.answer = option.to_sym
+          end
+
+          def self.included(base)
+            base.class_eval do
+              dbus_interface QUESTION_INTERFACE do
+                dbus_reader :id, "u"
+                dbus_reader :text, "s"
+                dbus_reader :options, "as"
+                dbus_reader :default_option, "s"
+                dbus_accessor :answer, "s"
+              end
+            end
+          end
+        end
+
+        # Interface to request a LUKS password
+        module LuksPassword
+          LUKS_PASSWORD_INTERFACE = "org.opensuse.DInstaller.Question.LuksPassword1"
+          private_constant :LUKS_PASSWORD_INTERFACE
+
+          # Given password
+          #
+          # @return [String]
+          def luks_password
+            backend.password || ""
+          end
+
+          # Sets a password
+          #
+          # @param value [String]
+          def luks_password=(value)
+            backend.password = value
+          end
+
+          def self.included(base)
+            base.class_eval do
+              dbus_interface LUKS_PASSWORD_INTERFACE do
+                dbus_accessor :luks_password, "s", dbus_name: "Value"
+              end
+            end
+          end
+        end
+      end
+
+      # Constructor
+      #
+      # @param backend [DInstaller::Question]
+      # @param logger [Logger]
+      def initialize(backend, logger)
+        @backend = backend
+        @logger = logger
+
+        super(build_path)
+
+        add_interfaces
+      end
+
+    private
+
+      # @return [DInstaller::Question]
+      attr_reader :backend
+
+      # Builds the question path under {DBus::Questions} path (e.g., o.o.DInstaller/Questions1/1)
+      #
+      # @return [String]
+      def build_path
+        Pathname.new(Questions.path).join(backend.id.to_s).to_s
+      end
+
+      # Adds interfaces to the question
+      def add_interfaces
+        interfaces.each { |i| singleton_class.include(i) }
+      end
+
+      # Interfaces the question implements
+      #
+      # @return [Array<Module>]
+      def interfaces
+        interfaces = [Interfaces::Question]
+        interfaces << Interfaces::LuksPassword if backend.is_a?(DInstaller::LuksQuestion)
```

Comment:
To be honest I do not like this part as it means that for each specialized question you need define it also here, so it is spread over multiple files. Can it live more closely? or at least add such mapping as constant to a bit more prominent place and document steps that needs to be done to have own question.